source/module_esolver/opt/test/CMakeLists.txt

```diff
@@ -0,0 +1,21 @@
+cmake_minimum_required(VERSION 3.20)
+
+project(CG_test)
+project(TN_test)
```

Comment:
Why do we need a independent CMake project here?

Reply:
For tests, please refer to our [docs](https://github.com/deepmodeling/abacus-develop/blob/develop/docs/CONTRIBUTING.md#adding-a-unit-test).

Reply:
I see, CG and TN have not been used in ABACUS for now, so I add this CMake project to compile the files in /source/module_esolver/opt separately. I will remove it and modify the tests to standard unittest when OFDFT module is completed.